docs/tileentities/tileentity.md

```diff
@@ -9,23 +9,22 @@ More advanced examples exist in mods, such as quarries, sorting machines, pipes,
     `TileEntities` aren't a solution for everything and they can cause lag when used wrongly.
     When possible, try to avoid them.
 
-## Creating a `TileEntity`
+## Registering
+
+Tile Entities are created and removed dynamically and as such are not registry objects on their own. 
+
+In order to create a `TileEntity`, you need to extend the `TileEntity` class. As such, another object is registered instead to easily create and refer to the *type* of the dynamic object. For a `TileEntity`, these are known as `TileEntityType`s.
+
+A `TileEntityType` can be [registered][registration] like any other registry object. To construct a `TileEntityType`, its builder form can be used via `TileEntityType$Builder#of`. This takes in two arguments: a `factory` representing some supplied method to create a new instance of the associated `TileEntity`, and a varargs of `validBlocks` which this `TileEntity` can be attached to. Building the `TileEntityType` is done by calling `TileEntityType$Builder#build`. This takes in a `Type` which represents the type-safe reference used to refer to this registry object in a `DataFixer`. Since `DataFixer`s are an optional system to use for mods, this can be passed as `null`.
 
-In order to create a `TileEntity`, you need to extend the `TileEntity` class.
-To register it, listen for the appropriate registry event and create a `TileEntityType`:
 ```java
-@SubscribeEvent
-public static void registerTE(RegistryEvent.Register<TileEntityType<?>> evt) {
-  TileEntityType<?> type = TileEntityType.Builder.of(factory, validBlocks).build(null);
-  type.setRegistryName("mymod", "myte");
-  evt.getRegistry().register(type);
-}
+// For some DeferredRegister<TileEntityType<?>> REGISTER
+public static final RegistryObject<TileEntityType<MyTE>> MY_TE = REGISTER.register("myte", () -> TileEntityType.Builder.of(factory, validBlocks).build(null));
```

Comment:
Could we have an admonition note to inform why a `null` is passed to the `#build` function?

Reply:
Read the last two lines of the explanation above the example. Should I move that down instead?